lib/spack/spack/build_environment.py

```diff
@@ -1020,6 +1021,15 @@ def set_all_package_py_globals(self):
                 pkg.setup_dependent_package(dependent_module, spec)
                 dependent_module.propagate_changes_to_mro()
 
+        for dspec, flag in chain(self.external, self.nonexternal):
```

Comment:
This should not execute on dependencies.

Reply:
Were they not previously? I took this out of a loop over the dependencies.